### ISSUE TITLE
serverless-function plugin does not work with form file params

### ISSUE THREAD

**Issue by wshirey** [Kong/kong#4106]
### Summary

[`serverless-plugin` docs example](https://docs.konghq.com/hub/kong-inc/serverless-functions) does not work as expected with 1.0rc3. Specifying a file form param doesn't work with cURL or Postman.

This seems similar to #3587 but there have been code modifications to this plugin since that issue was resolved.

### Steps To Reproduce

1. Follow steps 1-4 on docs page.
2. Step 5 as written returns

```
{
    "message": "schema violation (config.functions: expected an array)",
    "name": "schema violation",
    "fields": {
        "config": {
            "functions": "expected an array"
        }
    },
    "code": 2
}
```

Request works if you change `config.functions` to `config.functions[]`, but now functions is an empty array

```
curl -i -X POST http://localhost:8001/services/plugin-testing/plugins \
     -F "name=pre-function" \
     -F "config.functions[]=@custom-auth.lua"
HTTP/1.1 100 Continue

HTTP/1.1 201 Created
Date: Mon, 17 Dec 2018 20:53:12 GMT
Content-Type: application/json; charset=utf-8
Connection: keep-alive
Access-Control-Allow-Origin: *
Server: kong/1.0.0rc3
Content-Length: 246

{"created_at":1545079992,"config":{"functions":[]},"id":"883a8f5f-a1dc-47e9-a35a-f46d3fd37e44","service":{"id":"bac3e096-7463-4565-9c99-fac972efa3cb"},"enabled":true,"consumer":null,"run_on":"first","name":"pre-function","route":null,"api":null}
```
3. Alternatively, setting the script as text works in form or json body
```
curl -X POST \
  http://localhost:8001/services/plugin-testing/plugins \
  -H 'Content-Type: application/json' \
  -d '{
	"name": "pre-function",
	"config": {
		"functions": ["local a=kong.request.get_header('\''x-custom-auth'\'')if not a then return kong.response.exit(401,'\''Invalid Credentials'\'')end;kong.service.request.clear_header('\''x-custom-auth'\'')"]
	}
}'

{
    "created_at": 1545080401,
    "config": {
        "functions": [
            "local a=kong.request.get_header('x-custom-auth')if not a then return kong.response.exit(401,'Invalid Credentials')end;kong.service.request.clear_header('x-custom-auth')"
        ]
    },
    "id": "64cb01ab-3154-46dc-82d3-8dcdab5de88e",
    "service": {
        "id": "bac3e096-7463-4565-9c99-fac972efa3cb"
    },
    "enabled": true,
    "consumer": null,
    "run_on": "first",
    "name": "pre-function",
    "route": null,
    "api": null
}

```

### Additional Details & Logs

- Kong version `1.0.0rc3`
- Environment: `KONG_DOCKER_TAG=kong:1.0rc3 docker-compose up`

**Comment by kidd** [comment]
Thanks for the report @wshirey.  We got a fix in 1.0.0.

Stay tunned for the release, which should happen during today.

**Comment by p0pr0ck5** [comment]
@wshirey are you able to test this in the most recent release of Kong (1.0)?

**Comment by p0pr0ck5** [comment]
Closing this as stale and fixed in 1.0. If there are any more questions regarding this/buggy behavior in the most recent releases of kong, please open a new issue. Thanks!
